### DISCUSSION TITLE
What would be the best graphics library for a browser rendering engine?

### DISCUSSION THREAD

**Discussion by avrame** [vlang/v#19511] [Questions and Answers] [answered]
I'm following [this tutorial](https://limpet.net/mbrubeck/2014/08/08/toy-layout-engine-1.html) to build a toy browser. If I have the time I might keep working on it to add more functionality than the tutorial has. Anyway, I'm wondering what would be the best library to use for rendering? I'm thinking it would be SDL2, but maybe there's something better that the V community could suggest?

**Comment by medvednikov** [comment, marked as answer]
Hey,

Just the builtin `gg` would be enough.

It's very powerful, has text rendering, and uses Sokol, which gives you DX, GL, Metal.

**Reply by avrame** [reply]
Ok, thanks!
